src/main/scala/edu/ie3/mobsim/MobilitySimulator.scala

```diff
@@ -574,6 +582,20 @@ final class MobilitySimulator(
 
     timeUntilNextEvent
   }
+
+  private def updateElectricVehicles(movements: Seq[Movement]): Unit = {
+    val movementMap: Map[UUID, ElectricVehicle] = movements.map { movement =>
```

Comment:
I like building this map here, very efficient :)
Just a minor point: You don't need to assign types (`Map[UUID, ElectricVehicle]` here) to local vals.